CMakeLists.txt

```diff
@@ -169,12 +170,14 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
       target_link_libraries(snmalloc_lib INTERFACE "-rdynamic")
     endif()
 
-    if((${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64") OR
-       (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86") OR
-       (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm"))
-      check_cxx_compiler_flag(-march=native SUPPORT_MARCH_NATIVE)
-      if (SUPPORT_MARCH_NATIVE)
-        add_compile_options(-march=native)
+    if(SNMALLOC_OPTIMISE_FOR_CURRENT_MACHINE)
+      if((${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64") OR
```

Comment:
I think that if the user explicitly requests architecture-specific optimization, they should get it regardless of the system processor.